src/GraphQL/Resolvers/FuncFieldModelBinderResolver.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using GraphQL.Types;
+
+namespace GraphQL.Resolvers
+{
+    public class FuncFieldModelBinderResolver<TSourceType, TReturnType> : IFieldResolver<Task<TReturnType>>
+    {
+        private readonly Delegate _resolver;
+
+        public FuncFieldModelBinderResolver(Delegate resolver)
+        {
+            if (resolver == null)
+            {
+                throw new ArgumentNullException("A resolver function must be specified");
+            }
+            _resolver = resolver;
+        }
+
+        public async Task<TReturnType> Resolve(ResolveFieldContext context)
+        {
+            var parameters = RuntimeReflectionExtensions.GetMethodInfo(_resolver).GetParameters();
+
+            int index = 0;
+            var arguments = new object[parameters.Length];
+
+            if (context.As<TSourceType>().GetType() == parameters[0].ParameterType)
+            {
+                arguments[index] = context.As<TSourceType>();
+                index++;
+            }
+
+            foreach (var parameter in parameters.Skip(index))
+            {
+                arguments[index] = context.GetArgument<object>(parameter.Name);
```

Comment:
I think this should be changed to actually pass the `Type` of the parameter so it could cast/convert it properly.  You would need to refactor `GetArgument` so you could pass in a `Type`, and have `GetArgument<T>` delegate to that new method.